src/main/kotlin/com/ritense/iko/aggregateddataprofile/camel/AggregatedDataProfileRoute.kt

```diff
@@ -1,26 +1,75 @@
 package com.ritense.iko.aggregateddataprofile.camel
 
+import com.fasterxml.jackson.databind.JsonNode
 import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
 import com.ritense.iko.aggregateddataprofile.repository.AggregatedDataProfileRepository
 import org.apache.camel.builder.RouteBuilder
+import org.apache.camel.model.rest.ParamDefinition
+import org.apache.camel.model.rest.RestParamType.query
+import org.springframework.data.domain.Pageable
 
 class AggregatedDataProfileRoute(
-    val aggregatedDataProfileRepository: AggregatedDataProfileRepository,
-    val objectMapper: ObjectMapper,
+    private val aggregatedDataProfileRepository: AggregatedDataProfileRepository,
+    private val objectMapper: ObjectMapper,
 ) : RouteBuilder() {
     override fun configure() {
+        val containerParamsParamDefinition = ParamDefinition()
+            .name("containerParam")
+            .description("Container parameters for ADP")
+            .type(query)
+            .arrayType("String")
+            .required(false)
+
         rest("/aggregated-data-profiles")
+            .description("Resolve ADP by profile name")
+            .get("/{iko_profile}")
+            .param(containerParamsParamDefinition)
+            .to("direct:rest-to-adp")
             .get("/{iko_profile}/{iko_id}")
-            .to("direct:aggregated_data_profile_rest")
+            .param(containerParamsParamDefinition)
+            .to("direct:rest-to-adp-with-id")
+
+        from("direct:rest-to-adp")
+            .to("direct:aggregated-data-profile-container-params")
+
+        from("direct:rest-to-adp-with-id")
+            .to("direct:aggregated-data-profile-container-params")
+
+        from("direct:aggregated-data-profile-container-params")
+            .routeId("aggregated-data-profile-container-params")
+            .process { exchange ->
+                val containerParams: List<ContainerParam> = when (
+                    val paramHeader = exchange.`in`.getHeader("containerParam")
+                ) {
+                    is List<*> -> paramHeader.map { objectMapper.readValue(it.toString()) }
+                    is String -> listOf(objectMapper.readValue(paramHeader))
+                    else -> emptyList()
+                }
+                val endpointTransformContext: JsonNode = objectMapper
+                    .valueToTree(
+                        mapOf(
+                            "sortParams" to
+                                containerParams
+                                    .filter { !Pageable.unpaged().equals(it.pageable) }
```

Comment:
Checking the value against a unpaged.equals seems good. It just raised a question in my head. I assume this scenario is only when a container param is provided with only filter specified meaning the default of pageable is set?